File: aoc2023/test/aoc2023/day03_test.clj
(ns aoc2023.day03-test
  (:require [clojure.test :refer :all]
            [clojure.java.io]
            [aoc2023.day03 :as day03]))

(def sample-input (slurp (clojure.java.io/resource "day03-sample.txt")))

(deftest test-part1
  (testing "Day 3 Part 1"
    (is (= "4361" (day03/part1 sample-input)))
  )
)

;; (deftest test-part2
;;   (testing "Day 3 Part 2"
;;     (is (= 2286 (day01/part2 sample-input)))
;;   )
;; )

File: 2024/day03/day03_test.go
package day03

import (
	"fmt"
    "testing"
    )

var sample = "sample.txt"
var sample2 = "sample2.txt"

func TestPart1(t *testing.T) {
  result := Part1(sample)
  if result != 161 {
    fmt.Println(result)
  	t.Fail()
  }
}

func TestPart2(t *testing.T) {
  result := Part2(sample2)
  if result != 48 {
    fmt.Println(result)
  	t.Fail()
  }
}
